integrations/SnapEngage/browser.js

```diff
@@ -0,0 +1,202 @@
+            this.readyState === "loaded" ||
+            this.readyState === "complete")
+        ) {
+          done = true;
+          /* Place your SnapEngage JS API code below */
+          /* SnapEngage.allowChatSound(true); Example JS API: Enable sounds for Visitors. */
+        }
+      };
+      const s = document.getElementsByTagName("script")[0];
+      s.parentNode.insertBefore(se, s);
+    })(this.widgetId);
+  }
+
+  init() {
+    logger.debug("===In init SnapEngage===");
+    this.loadScript();
+  }
+
+  isLoaded() {
+    logger.debug("===In isLoaded SnapEngage===");
+    return !!window.SnapEngage;
+  }
+
+  isReady() {
+    logger.debug("===In isReady SnapEngage===");
+
+    // Dasboard Other Settings
+    if (this.recordLiveChatEvents) {
+      const standardEventsMap = getHashFromArray(this.eventsToStandard);
+      if (
+        Object.keys(standardEventsMap).length === 0 &&
+        standardEventsMap.constructor === Object
+      ) {
+        this.recordingLiveChatEvents();
+      } else {
+        this.recordingLiveChatEventsWithMapping(
+          this.updateEventNames,
+          standardEventsMap
+        );
+      }
+    }
+    return !!window.SnapEngage;
+  }
+
+  identify(rudderElement) {
+    logger.debug("===In SnapEngage Identify===");
+    const { message } = rudderElement;
+    const email =
+      get(message, "context.traits.email") || get(message, "traits.email");
+
+    if (!email) {
+      logger.error("User parameter (email) is required for identify call");
+      return;
+    }
+    window.SnapEngage.setUserEmail(email);
+
+    const name =
+      get(message, "context.traits.name") || get(message, "traits.name");
+
+    if (name) {
+      window.SnapEngage.setUserName(name);
+    }
+  }
+
+  recordingLiveChatEvents() {
+    window.SnapEngage.setCallback("StartChat", function () {
+      window.rudderanalytics.track(
+        "Live Chat Conversation Started",
+        {},
+        { context: { integration: integrationContext } }
+      );
+    });
+
+    window.SnapEngage.setCallback("ChatMessageReceived", function (agent) {
+      window.rudderanalytics.track(
+        "Live Chat Message Received",
+        { agentUsername: agent },
+        { context: { integration: integrationContext } }
+      );
+    });
+
+    window.SnapEngage.setCallback("ChatMessageSent", function () {
+      window.rudderanalytics.track(
+        "Live Chat Message Sent",
+        {},
+        { context: { integration: integrationContext } }
+      );
+    });
+
+    window.SnapEngage.setCallback("Close", function () {
+      window.rudderanalytics.track(
+        "Live Chat Conversation Ended",
+        {},
+        { context: { integration: integrationContext } }
+      );
+    });
+
+    window.SnapEngage.setCallback("InlineButtonClicked", function () {
+      window.rudderanalytics.track(
+        "Inline Button Clicked",
+        {},
+        { context: { integration: integrationContext } }
+      );
+    });
```

Comment:
Can you check if we are getting any other fields with this function or something similar to `ChatMessageReceived`?

Reply:
We can do it as an enhancement.